src/button/index.js

```diff
@@ -1,19 +1,31 @@
-import ButtonPrimary from './button-primary';
-import ButtonSecondary from './button-secondary';
-import ButtonWhite from './button-white';
-import ButtonPill from './button-pill';
-import ButtonWarning from './button-warning';
-import ButtonOcean from './button-ocean';
-import ButtonDark from './button-dark';
-import LoadingIcon from './button-loading';
+import React from 'react';
+import StyledButton from './style';
+import PropTypes from 'prop-types';
+import Loading from './loading.svg';
 
-export {
-  ButtonPrimary,
-  ButtonSecondary,
-  ButtonWhite,
-  ButtonPill,
-  ButtonWarning,
-  ButtonOcean,
-  ButtonDark,
-  LoadingIcon
+const Button = props => {
+  const { children, loading } = props;
+  return (
+    <StyledButton {...props}>
+      {children}
+      {loading && <Loading />}
+    </StyledButton>
+  );
 };
+
+Button.propTypes = {
+  color: PropTypes.oneOf(['primary', 'secondary', 'warning', 'success']),
+  children: PropTypes.any.isRequired,
```

Comment:
Is this basically text/svg?

Reply:
Yes, that's correct. I wasn't entirely sure how to support both. The other option I thought of was passing down an icon prop and then children would always be text.

Reply:
I think this is fine, icon in the props is fine too.

Reply:
So this is proving a little harder than I expected with the SVG loader and passing it down. I'll move this to a later feature.